datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -49,6 +53,62 @@ use super::coercion::{
     eq_coercion, like_coercion, numerical_coercion, order_coercion, string_coercion,
 };
 
+// Simple (low performance) kernels until optimized kernels are added to arrow
+// TODO: file arrow-rs ticket to track this feature
```

Comment:
Question to reviewers: Are these useful kernels to have in arrow? Or are they so special purpose we should leave them just in DataFusion?

Reply:
I didn't know about the `is [not] distinct from` SQL comparison before reading this PR, but it seems to be somewhat standard:
![image](https://user-images.githubusercontent.com/7913347/142585062-44b0e9e6-c77d-48e4-b428-770ba9f75fe5.png)
So I guess that having kernels for it makes sense.

Reply:
```suggestion
// See https://github.com/apache/arrow-rs/issues/960
```

Reply:
filed https://github.com/apache/arrow-rs/issues/960